src/go/rdctl/cmd/start.go

```diff
@@ -0,0 +1,234 @@
+		}
+		rdctlPath, err := os.Executable()
+		if err != nil {
+			rdctlPath = ""
+		}
+		applicationPath = getPathFunc(rdctlPath)
+		if applicationPath == "" {
+			return fmt.Errorf("could not locate main Rancher Desktop executable; please retry with the --path option")
+		}
+	}
+	return launchApp(applicationPath, commandLineArgs)
+}
+
+func launchApp(applicationPath string, commandLineArgs []string) error {
+	var commandName string
+	var args []string
+
+	if runtime.GOOS == "darwin" {
+		commandName = "/usr/bin/open"
+		args = []string{"-a", applicationPath}
+		if len(commandLineArgs) > 0 {
+			args = append(args, "--args")
+			args = append(args, commandLineArgs...)
+		}
+	} else {
+		commandName = applicationPath
+		args = commandLineArgs
+	}
+	// Include this output because there's a delay before the UI comes up.
+	// Without this line, it might look like the command doesn't work.
+	fmt.Fprintf(os.Stderr, "About to launch %s %s ...\n", commandName, strings.Join(args, " "))
+	cmd := exec.Command(commandName, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func moveToParent(fullPath string, numberTimes int) string {
+	fullPath = path.Clean(fullPath)
+	for ; numberTimes > 0; numberTimes-- {
+		fullPath = path.Dir(fullPath)
+	}
+	return fullPath
+}
+
+func getWindowsRDPath(rdctlPath string) string {
+	if rdctlPath != "" {
+		normalParentPath := moveToParent(rdctlPath, 5)
+		candidatePath := checkExistence(path.Join(normalParentPath, "Rancher Desktop.exe"), 0)
+		if candidatePath != "" {
+			return candidatePath
+		}
+	}
+	localAppDataGetters := []func() string{
+		func() string {
+			return os.Getenv("LOCALAPPDATA")
+		},
+		func() string {
+			homePath, err := os.UserHomeDir()
+			if err != nil || homePath == "" {
+				return ""
+			}
+			return path.Join(homePath, "AppData", "Local")
+		},
+		func() string {
+			homeDrive := os.Getenv("HOMEDRIVE")
+			if homeDrive == "" {
+				return ""
+			}
+			homePathPart := os.Getenv("HOMEPATH")
+			if homePathPart == "" {
+				return ""
+			}
+			return path.Join(homeDrive+homePathPart, "AppData", "Local")
+		},
+		func() string {
+			homePath := os.Getenv("HOME")
+			if homePath == "" {
+				return ""
+			}
+			return path.Join(homePath, "AppData", "Local")
+		},
+	}
+	for _, localAppDataGetter := range localAppDataGetters {
+		localAppDataDir := localAppDataGetter()
+		candidatePath := checkExistence(path.Join(localAppDataDir, "Programs", "Rancher Desktop", "Rancher Desktop.exe"), 0)
+		if candidatePath != "" {
+			return candidatePath
+		}
+	}
+	return ""
+}
+
+func getMacOSRDPath(rdctlPath string) string {
+	if rdctlPath != "" {
+		// we're at .../Applications/R D.app (could have a different name)/Contents/Resources/resources/darwin/bin
+		// and want to move to the "R D.app" part
+		RDAppParentPath := moveToParent(rdctlPath, 6)
+		if checkExistence(path.Join(RDAppParentPath, "Contents", "MacOS", "Rancher Desktop"), 111) != "" {
```

Comment:
```suggestion
		if checkExistence(path.Join(RDAppParentPath, "Contents", "MacOS", "Rancher Desktop"), 0o111) != "" {
```
Presumably this is meant to be octal

Reply:
yes :)